openssl/src/x509/mod.rs

```diff
@@ -390,6 +390,69 @@ impl X509Generator {
         }
     }
 
+    /// Sets the certificate public-key, then self-sign and return it
```

Comment:
It seems like a good bit of the code between this and sign (and maybe others?) can be shared.